airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -287,6 +275,12 @@ def __init__(
         self.pod_request_obj: k8s.V1Pod | None = None
         self.pod: k8s.V1Pod | None = None
 
+        if "resources" in kwargs:
```

Comment:
why not  keep the exact same check? i.e.
`if isinstance(resources, k8s.V1ResourceRequirements):`

i don't even know what resources are for.... but... is it really true that they should never be used for KPO?

Reply:
but i guess the idea was simply to catch that case you mentioned, when user _thinks_ they are still supplying container requirements but they are instead going to base operator. seems checking the obj type like this would cover that case, while still allowing normal usage of `resources` -- whatever that is... though maybe i'm missing some pieces

Reply:
BaseOperator resources is used to customized cpu/memory for a task.
Think about a case where you execute PythonOperator but you want to limit it to specific memory limit of the worker.
In KPO there is no effect (to my understanding) of using BaseOperator resources since the resource allocation is against the Kubernetes cluster.

BaseOperator resources is dict which is creating a [Resource](https://github.com/apache/airflow/blob/fd859d138ce9558e19d73cd0cbbe9ef3d7993b64/airflow/utils/operator_resources.py#L31) object

To answer your question:
`if isinstance(resources, k8s.V1ResourceRequirements):`

Is not case i'm worried about. If user will provide `V1ResourceRequirements` for `resources` - his code will not work since `resources` expecting dict
https://github.com/apache/airflow/blob/fd859d138ce9558e19d73cd0cbbe9ef3d7993b64/airflow/models/baseoperator.py#L733

The issue I'm trying to protect against is when user is passing K8s resource allocation as dict via the `resource` parameter.  This is something the user could have done until now because we backport and convert the dict to V1ResourceRequirements but from now on it will not work.
The issue is that if user didn't notice the change log and continue to pass resources with K8s parameters like in https://stackoverflow.com/questions/73959192/airflow-kubernetespodoperator-broken-dag-unexpected-keyword-argument-reque 

WDYT?

Reply:
BTW possible BaseOperator resources is never used anyway?
Ash mentioned it in https://github.com/apache/airflow/issues/16563#issuecomment-865045209

Reply:
i really do think it's probably best to keep the same if but just raise instead of warn (and not raise whenever resources passed)

1. it's kind of us to raise in this case because clearly user thinks they are setting k8s resources but it's not gonna work anymore so we should absolutely raise in this case
2. it's unnecessary to raise otherwise because, whether useless or not, it's still a valid base operator param so really, there's no reason not to allow it.  remember that even with KPO, the task itself may run on celery (or whatever executor) -- the "pod" will get the container_resources but the task itself will not.

i realize, this is not of great consequence, and you have my approval anyway, just calling it like i see it.

Reply:
I am not saying we would not remove the resource parameter.  That's a separate topic.  What we're doing here is, we're no longer helping users out by converting `resources` into `container_resources`.  In case user doesn't catch this change (as evidenced by v1resourcerequirements being passed to resources) we raise now instead of warn.  But user could still in theory use `resources` for whatever it's there for (i'm assuming that's something other than v1resourcerequirements).

Essentially, the reason we should raise _conditionally_, based on what is passed to resources, is precisely the same as whatever motivated us to _warn_ conditionally -- if there was a reason to do that there (and i'm assuming that is to allow "proper" use of resources param), then there's a reason to do that here.

And then maybe next major release we remove the check and just don't even look at resources.

Whether resources gets chopped from airflow, that's a separate question.